### PR TITLE
feat: support both controlled and uncontrolled for tab

### PR DESCRIPTION
**Pull request by dangbt**
- change uncontrol tab to control tab
- add callback when changing tab

### CODE REVIEW

File: core/src/tab/tab.stories.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { storiesOf } from "@storybook/react"; //eslint-disable-line
-import React from "react";
+import React, { useState } from "react";
```
Comment (thien-do) [inline core/src/tab/tab.stories.tsx:2]: We already imported "React" so it's better to just use React.useState

Also it should be import * as React

File: core/src/tab/tab.stories.tsx [hunk 2]
```diff
@@ -8,18 +8,21 @@ const tabs: Tab[] = [
...
-		<DivPx size={16} />
-		<Tabs style={Tabs.styles.flat} children={tabs} />
-		<DivPx size={16} />
-		<Tabs style={Tabs.styles.flat} children={tabs} noPadding />
-		<DivPx size={16} />
-		<div style={{ height: 200 }}>
-			<Tabs children={tabs} fullHeight />
+storiesOf("Tab", module).add("Primary", () => {
+	const [activeTab, setActiveTab] = useState('Second')
+	return (
```
Comment (thien-do) [inline core/src/tab/tab.stories.tsx:13]: Should not replace the whole story. Should add another one for controlled case, while keeping the uncontrolled case

File: core/src/tab/tab.tsx [hunk 3]
```diff
@@ -21,57 +21,56 @@ interface TabStyle {
 
 interface Props {
 	children: Tab[];
+	activeTab: string;
+	setActiveTab: (tab: string) => void;
 	initialTab?: string;
 	noPadding?: boolean;
 	style?: TabStyle;
 	fullHeight?: boolean;
+	callbackOnTab?: (tabId: string) => void;
```
Comment (thien-do) [inline core/src/tab/tab.tsx:30]: In which case should we use this?

File: core/src/tab/tab.tsx [hunk 4]
```diff
@@ -21,57 +21,56 @@ interface TabStyle {
 
 interface Props {
 	children: Tab[];
+	activeTab: string;
+	setActiveTab: (tab: string) => void;
```
Comment (thien-do) [inline core/src/tab/tab.tsx:25]: These should not be required. We should support both controlled and uncontrolled use case

**Review by thien-do — CHANGES_REQUESTED** [review]
Should support both controlled and uncontrolled use cases

File: core/src/tab/tab.tsx (hunk 3 above)
Comment (dangbt) [reply core/src/tab/tab.tsx:32]: In this case, if we want to callback a function when clicking the tab. Ex, we want to change the query of URL

**Pull request merged**